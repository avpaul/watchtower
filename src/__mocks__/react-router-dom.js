import React from 'react';
import PropTypes from 'prop-types';

const rrd = require('react-router-dom');

// Just render plain div with its children
rrd.BrowserRouter = ({ children }) => <div>{children}</div>;

rrd.BrowserRouter.propTypes = {
  children: PropTypes.objectOf(PropTypes.element, PropTypes.func).isRequired,
};

module.exports = rrd;
