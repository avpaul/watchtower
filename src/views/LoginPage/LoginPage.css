.login-page {
  width: 100%;
  padding: 1rem;
  background-color: #1d66de;
  background-image: url(../../static/Landing-Image.png);
  background-size: 800px 450px;
  background-repeat: no-repeat;
  min-height: 100vh;
  background-position: 90% 100%;
}

.toaster {
  background-color: #ffa550;
  color: white;
}

.login-page__logo {
  padding: 15% 5% 0 25%;
}

.login-page__tagline {
  font-size: 18px;
  line-height: 22px;
  text-align: center;
  margin: 10% 5%;
  color: #ffffff;
}

.login-page__btn-container {
  margin: 10% auto;
  margin-left: 15%;
}

.login-page__btn {
  border-radius: 8px;
  background-color: #ffffff;
  color: #3359db;
  font-family: 'DIN Pro';
  font-size: 1rem;
  line-height: 22px;
  padding: 4%;
  box-shadow: 0 5px 12px 0 rgba(83, 89, 98, 0.12);
  cursor: pointer;
}

.login-page__btn-logo {
  height: 15px;
  padding-right: 4%;
  border-radius: 4px 0 0 4px;
  background-color: #ffffff;
}

.login-page__btn-text {
  padding-left: 1.5%;
  color: #000;
}

.login-page__btn:hover {
  text-decoration: none;
}

@media only screen and (min-width: 425px) {
  .login-page__btn-container {
    margin-left: 25%;
  }

  .login-page {
    background-position: 100% 100%;
  }
}

@media only screen and (min-width: 450px) {
  .page-info__container {
    width: 100%;
    margin: 9% 2%;
  }

  .login-page__tagline {
    font-size: 1.2rem;
    text-align: center;
    margin: 3% 5%;
  }
  .login-page__btn-container {
    width: 50%;
    margin: 15% auto;
  }

  .login-page__btn {
    padding: 4%;
  }
  .login-page__btn-logo {
    padding-right: unset;
  }
  .login-page__btn-text {
    padding-left: 1.5%;
  }
}

@media only screen and (min-width: 700px) {
  .page-info__container {
    width: 100%;
    margin: 9% -15%;
  }
  .login-page__tagline {
    font-size: 1.5rem;
  }
  .login-page__btn-container {
    margin: 10% auto;
  }
}

@media only screen and (min-width: 801px) {
  .login-page {
    background-size: 1000px 800px;
    background-position: 80% -120%;
  }

  .page-info__container {
    margin: 9% -22%;
  }
}

@media only screen and (min-width: 1000px) {
  .login-page {
    background-size: 1440px 826px;
    background-position: 100% -140%;
  }

  .page-info__container {
    width: 60%;
    margin: 10% -5%;
  }

  .login-page__tagline {
    font-size: 2rem;
    line-height: 2.5rem;
  }

  .login-page__btn-text {
    height: 41px;
    width: 160px;
    color: #000000d6;
    font-family: 'DIN Pro';
    font-size: 18px;
    line-height: 40px;
    padding-left: 2%;
    padding-right: 2%;
  }

  .login-page__btn-logo {
    height: 21px;
    padding-left: 1.5%;
    margin-bottom: 0.5%;
  }
  .login-page__btn-container {
    margin-top: 5%;
    margin-left: 20%;
  }
}

@media only screen and (min-width: 1400px) {
  .login-page__tagline {
    height: 158px;
    width: 507px;
    color: #ffffff;
    font-family: 'DIN Pro';
    font-size: 45px;
    font-weight: 600;
    line-height: 66px;
    letter-spacing: 1.5px;
    margin-bottom: -3%;
  }

  .login-page__header {
    margin-top: -13%;
  }

  .page-info__container {
    width: 100%;
    margin-left: 10%;
  }
  .login-page__tagline {
    text-align: left;
  }
  .login-page__logo {
    padding-left: 5%;
    margin-bottom: 3%;
  }

  .login-page__btn {
    padding: 2%;
  }
  .login-page__btn-container {
    margin-left: 5%;
  }
}
