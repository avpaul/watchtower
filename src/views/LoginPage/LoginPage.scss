@import '../../assets/sass/abstract/variables';

.login-page {
  width: 100%;
  padding: 1rem;
  background-color: #1d66de;
  background-image: url(../../static/Landing-Image.png);
  background-size: 700px 450px;
  background-repeat: no-repeat;
  min-height: 100vh;
  background-position: 95% 100%;
  &__logo {
    padding: 15% 5% 0 25%;
  }

  &__tagline {
    font-size: 18px;
    line-height: 22px;
    margin: 10% 5%;
    color: #ffffff;
    text-align: left;
  }

  &__btn {
    border-radius: 8px;
    background-color: #ffffff;
    color: #3359db;
    font-family: $font-primary;
    font-size: 1rem;
    line-height: 22px;
    padding: 4%;
    box-shadow: 0 5px 12px 0 rgba(83, 89, 98, 0.12);
    cursor: pointer;
    &-logo {
      height: 15px;
      padding-right: 4%;
      border-radius: 4px 0 0 4px;
      background-color: #ffffff;
    }

    &-text {
      padding-left: 1.5%;
      color: #000;
    }

    &:hover {
      text-decoration: none;
    }

    &-container {
      margin: 10% auto;
      margin-left: 15%;
    }
  }

  @media only screen and (min-width: 425px) {
    &__btn-container {
      margin-left: 25%;
    }

    & {
      background-position: 100% 100%;
    }
  }

  @media only screen and (min-width: 450px) {
    .page-info__container {
      width: 100%;
      margin: 9% 2%;
    }

    &__tagline {
      font-size: 1.2rem;
      text-align: center;
      margin: 3% 5%;
    }

    &__btn {
      padding: 4%;

      &-logo {
        padding-right: unset;
      }

      &-text {
        padding-left: 1.5%;
      }
      &-container {
        width: 50%;
        margin: 15% auto;
      }
    }
  }

  @media only screen and (min-width: 700px) {
    & {
      background-size: 1000px 750px;
      background-position: 100% 100%;
    }

    .page-info__container {
      width: 100%;
      margin: 9% -15%;
    }

    &__tagline {
      font-size: 2rem;
      line-height: 3.5rem;
      width: 50%;
      text-align: left;
      margin-left: 25%;
    }

    &__btn-container {
      margin: 10% auto;
    }
  }

  @media only screen and (min-width: 801px) {
    & {
      background-size: 1000px 800px;
      background-position: 80% -120%;
    }

    .page-info__container {
      margin: 9% -22%;
    }
  }

  @media only screen and (min-width: 1000px) {
    & {
      background-size: cover;
      background-position: center;
    }

    .page-info__container {
      width: 60%;
      margin: 10% -5%;
    }

    &__tagline {
      margin-left: 20%;
    }

    &__btn {
      &-text {
        height: 41px;
        width: 160px;
        color: #000000;
        font-family: $font-primary;
        font-size: 18px;
        line-height: 40px;
        padding-left: 2%;
        padding-right: 2%;
      }

      &-logo {
        height: 21px;
        padding-left: 1.5%;
        margin-bottom: 0.5%;
      }

      &-container {
        margin-top: 5%;
        margin-left: 20%;
      }
    }
  }

  @media only screen and (min-width: 1400px) {
    &__tagline {
      height: 158px;
      width: 507px;
      color: #ffffff;
      font-family: $font-primary;
      font-size: 45px;
      font-weight: 600;
      line-height: 66px;
      letter-spacing: 1.5px;
      margin-bottom: -3%;
      margin-left: 5%;
    }

    &__header {
      margin-top: -13%;
    }

    .page-info__container {
      width: 100%;
      margin-left: 10%;
    }

    &__logo {
      padding-left: 5%;
      margin-bottom: 3%;
    }

    &__btn {
      padding: 2%;
    }

    &__btn-container {
      margin-left: 5%;
    }
  }
}

.toaster {
  background-color: #ffa550;
  color: white;
}
