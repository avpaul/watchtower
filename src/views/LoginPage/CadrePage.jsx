import React from 'react';
import {Link} from 'react-router-dom';
import './CadrePage.css';
import illustration from '../../static/cadre-engineer-illustration.svg';

export default () => (
    <div className="cadre-engineer">
        <p className="hey-david">Hey, name!</p>
        <p className="welcome-to-the-cadr">
            Welcome, to Cadre Program{' '}
            <span role="img" className="welcome-to-the-cadr" aria-label="Celebration">
        ðŸŽ‰
      </span>
        </p>
        ,
        <img
            src={illustration}
            alt="cadre-illustration"
            className="cadreIllustration"
        />
        <p className="we-are-glad-you-made">
            We are glad you made it to the stage.Here, you&apos;d be able to select
            your Engineering path.
        </p>
        <button type="submit" className="rectangle">
            <Link className="cader-link" to="/dashboard">
                CONTINUE
            </Link>
        </button>
    </div>
);
