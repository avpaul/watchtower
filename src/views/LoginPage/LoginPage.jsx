import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import authService from '../../services/auth';
import andelaLogo from '../../static/Logo-andela.png';
import googleLogo from '../../static/icons8-google-48.png';
import watchTowerLogo from '../../static/Logo-watchTower.svg';
import './LoginPage.css';

/**
 * LoginPage UI Component
 *
 * @returns {JSX} React component
 */
class LoginPage extends Component {
  constructor(props) {
    super(props);
    const { authHostUrl } = this.props;
    const { authRedirectUrl } = this.props;

    this.state = {
      loggedIn: authService.isAuthenticated(),
      authUrl: `${authHostUrl}/login?redirect_url=${authRedirectUrl}`,
    };
    this.handleLogin = this.handleLogin.bind(this);
  }

  handleLogin(event) {
    event.preventDefault();
    const { authUrl } = this.state;
    window.location.replace(authUrl);
  }

  render() {
    const { loggedIn, authUrl } = this.state;
    if (loggedIn) {
      return (
        <Redirect to="/dashboard" />
      );
    }
    return (
      <main className=".container-fluid text-center login-page">
        <div className="login-page__header">
          <img
            src={watchTowerLogo}
            alt="watch-tower-logo"
            className="login-page__logo"
          />
          <span className="login-page__text-by">by</span>
          <img
            src={andelaLogo}
            alt="andela-logo"
            className="login-page__logo"
          />
        </div>
        <div className="login-page__tagline text-center">
          <p className="m-0">Tracking</p>
          <p>DevPulse Ratings & LMS Scores</p>
        </div>
        <div style={{ marginTop: '40px' }} title="Login with google">
          <a href={authUrl} onClick={this.handleLogin} className="login-page__btn" tabIndex={0}>
            <img src={googleLogo} alt="google-icon" className="login-page__btn-logo" />
            <span className="login-page__line" />
            <span className="login-page__btn-text">LOGIN WITH GOOGLE</span>
          </a>
        </div>
      </main>
    );
  }
}

LoginPage.propTypes = {
  authHostUrl: PropTypes.string,
  authRedirectUrl: PropTypes.string,
};

LoginPage.defaultProps = {
  authHostUrl: process.env.REACT_APP_ANDELA_AUTH_HOST,
  authRedirectUrl: process.env.REACT_APP_GOOGLE_AUTH_REDIRECT_URL,
};

export default LoginPage;
