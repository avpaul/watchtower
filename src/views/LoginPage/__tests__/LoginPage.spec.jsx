import React from 'react';
import { shallow } from 'enzyme';
import { Redirect } from 'react-router-dom';
import Cookie from 'js-cookie';
import LoginPage from '../LoginPage';

describe('<LoginPage />', () => {
  const ANDELA_AUTH_URL = 'TEST_AUTH_URL';
  const REDIRECT_URL = 'TEST_REDIRECT_URL';

  let wrapper;
  let loginButtonSpy;
  beforeAll(() => {
    loginButtonSpy = jest.spyOn(LoginPage.prototype, 'handleLogin');
    wrapper = shallow(<LoginPage
      authHostUrl={ANDELA_AUTH_URL}
      authRedirectUrl={REDIRECT_URL}
    />);
  });
  it('should render login button with correct text', () => {
    const loginButton = wrapper.find('.login-page__btn');
    expect(loginButton.text()).toEqual('LOGIN WITH GOOGLE');
  });
  it('should attempt sign-in when login button is clicked', () => {
    const loginButton = wrapper.find('.login-page__btn');
    loginButton.simulate('click', { preventDefault: jest.fn() });
    expect(loginButtonSpy).toHaveBeenCalled();
  });
  it('It redirects user to dashboard when logged in', () => {
    const fakeToken = 'sfdfjnsdfowbefijbmdnkf';
    Cookie.set('jwt-token', fakeToken, { path: '/' });
    const instance = shallow(<LoginPage
      authHostUrl={ANDELA_AUTH_URL}
      authRedirectUrl={REDIRECT_URL}
    />);
    expect(instance.find(Redirect).length).toBe(1);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
