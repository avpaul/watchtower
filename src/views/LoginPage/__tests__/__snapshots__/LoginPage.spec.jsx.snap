// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage
    authHostUrl="TEST_AUTH_URL"
    authRedirectUrl="TEST_REDIRECT_URL"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="login-page__header"
>
          <img
                    alt="watch-tower-logo"
                    className="login-page__logo"
                    src="Logo-watchTower.svg"
          />
          <span
                    className="login-page__text-by"
          >
                    by
          </span>
          <img
                    alt="andela-logo"
                    className="login-page__logo"
                    src="Logo-andela.png"
          />
</div>,
        <div
          className="login-page__tagline text-center"
>
          <p
                    className="m-0"
          >
                    Tracking
          </p>
          <p>
                    DevPulse Ratings & LMS Scores
          </p>
</div>,
        <div
          style={
                    Object {
                              "marginTop": "40px",
                            }
          }
          title="Login with google"
>
          <a
                    className="login-page__btn"
                    href="TEST_AUTH_URL/login?redirect_url=TEST_REDIRECT_URL"
                    onClick={[Function]}
                    tabIndex={0}
          >
                    <img
                              alt="google-icon"
                              className="login-page__btn-logo"
                              src="icons8-google-48.png"
                    />
                    <span
                              className="login-page__line"
                    />
                    <span
                              className="login-page__btn-text"
                    >
                              LOGIN WITH GOOGLE
                    </span>
          </a>
</div>,
      ],
      "className": ".container-fluid text-center login-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="watch-tower-logo"
              className="login-page__logo"
              src="Logo-watchTower.svg"
/>,
            <span
              className="login-page__text-by"
>
              by
</span>,
            <img
              alt="andela-logo"
              className="login-page__logo"
              src="Logo-andela.png"
/>,
          ],
          "className": "login-page__header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "watch-tower-logo",
              "className": "login-page__logo",
              "src": "Logo-watchTower.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "by",
              "className": "login-page__text-by",
            },
            "ref": null,
            "rendered": "by",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "andela-logo",
              "className": "login-page__logo",
              "src": "Logo-andela.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="m-0"
>
              Tracking
</p>,
            <p>
              DevPulse Ratings & LMS Scores
</p>,
          ],
          "className": "login-page__tagline text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tracking",
              "className": "m-0",
            },
            "ref": null,
            "rendered": "Tracking",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "DevPulse Ratings & LMS Scores",
            },
            "ref": null,
            "rendered": "DevPulse Ratings & LMS Scores",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="login-page__btn"
            href="TEST_AUTH_URL/login?redirect_url=TEST_REDIRECT_URL"
            onClick={[Function]}
            tabIndex={0}
>
            <img
                        alt="google-icon"
                        className="login-page__btn-logo"
                        src="icons8-google-48.png"
            />
            <span
                        className="login-page__line"
            />
            <span
                        className="login-page__btn-text"
            >
                        LOGIN WITH GOOGLE
            </span>
</a>,
          "style": Object {
            "marginTop": "40px",
          },
          "title": "Login with google",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="google-icon"
                className="login-page__btn-logo"
                src="icons8-google-48.png"
/>,
              <span
                className="login-page__line"
/>,
              <span
                className="login-page__btn-text"
>
                LOGIN WITH GOOGLE
</span>,
            ],
            "className": "login-page__btn",
            "href": "TEST_AUTH_URL/login?redirect_url=TEST_REDIRECT_URL",
            "onClick": [Function],
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "google-icon",
                "className": "login-page__btn-logo",
                "src": "icons8-google-48.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "login-page__line",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "LOGIN WITH GOOGLE",
                "className": "login-page__btn-text",
              },
              "ref": null,
              "rendered": "LOGIN WITH GOOGLE",
              "type": "span",
            },
          ],
          "type": "a",
        },
        "type": "div",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="login-page__header"
>
            <img
                        alt="watch-tower-logo"
                        className="login-page__logo"
                        src="Logo-watchTower.svg"
            />
            <span
                        className="login-page__text-by"
            >
                        by
            </span>
            <img
                        alt="andela-logo"
                        className="login-page__logo"
                        src="Logo-andela.png"
            />
</div>,
          <div
            className="login-page__tagline text-center"
>
            <p
                        className="m-0"
            >
                        Tracking
            </p>
            <p>
                        DevPulse Ratings & LMS Scores
            </p>
</div>,
          <div
            style={
                        Object {
                                    "marginTop": "40px",
                                  }
            }
            title="Login with google"
>
            <a
                        className="login-page__btn"
                        href="TEST_AUTH_URL/login?redirect_url=TEST_REDIRECT_URL"
                        onClick={[Function]}
                        tabIndex={0}
            >
                        <img
                                    alt="google-icon"
                                    className="login-page__btn-logo"
                                    src="icons8-google-48.png"
                        />
                        <span
                                    className="login-page__line"
                        />
                        <span
                                    className="login-page__btn-text"
                        >
                                    LOGIN WITH GOOGLE
                        </span>
            </a>
</div>,
        ],
        "className": ".container-fluid text-center login-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="watch-tower-logo"
                className="login-page__logo"
                src="Logo-watchTower.svg"
/>,
              <span
                className="login-page__text-by"
>
                by
</span>,
              <img
                alt="andela-logo"
                className="login-page__logo"
                src="Logo-andela.png"
/>,
            ],
            "className": "login-page__header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "watch-tower-logo",
                "className": "login-page__logo",
                "src": "Logo-watchTower.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "by",
                "className": "login-page__text-by",
              },
              "ref": null,
              "rendered": "by",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "andela-logo",
                "className": "login-page__logo",
                "src": "Logo-andela.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="m-0"
>
                Tracking
</p>,
              <p>
                DevPulse Ratings & LMS Scores
</p>,
            ],
            "className": "login-page__tagline text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tracking",
                "className": "m-0",
              },
              "ref": null,
              "rendered": "Tracking",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "DevPulse Ratings & LMS Scores",
              },
              "ref": null,
              "rendered": "DevPulse Ratings & LMS Scores",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="login-page__btn"
              href="TEST_AUTH_URL/login?redirect_url=TEST_REDIRECT_URL"
              onClick={[Function]}
              tabIndex={0}
>
              <img
                            alt="google-icon"
                            className="login-page__btn-logo"
                            src="icons8-google-48.png"
              />
              <span
                            className="login-page__line"
              />
              <span
                            className="login-page__btn-text"
              >
                            LOGIN WITH GOOGLE
              </span>
</a>,
            "style": Object {
              "marginTop": "40px",
            },
            "title": "Login with google",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="google-icon"
                  className="login-page__btn-logo"
                  src="icons8-google-48.png"
/>,
                <span
                  className="login-page__line"
/>,
                <span
                  className="login-page__btn-text"
>
                  LOGIN WITH GOOGLE
</span>,
              ],
              "className": "login-page__btn",
              "href": "TEST_AUTH_URL/login?redirect_url=TEST_REDIRECT_URL",
              "onClick": [Function],
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "google-icon",
                  "className": "login-page__btn-logo",
                  "src": "icons8-google-48.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "login-page__line",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "LOGIN WITH GOOGLE",
                  "className": "login-page__btn-text",
                },
                "ref": null,
                "rendered": "LOGIN WITH GOOGLE",
                "type": "span",
              },
            ],
            "type": "a",
          },
          "type": "div",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
